hbase_sql/src/main/java/com/splicemachine/stream/StreamableRDD.java

```diff
@@ -60,25 +63,36 @@
     private final int parallelPartitions;
     private OlapStatus jobStatus;
     private OlapStreamListener olapStreamListener;
+    private final int executorThreads;
 
 
+    /**
+     * A small constructor for the ITs with default parameters.
+     */
     StreamableRDD(JavaRDD<T> rdd, UUID uuid, String clientHost, int clientPort) {
         this(rdd, null, uuid, clientHost, clientPort, 2, 512);
     }
 
+    /**
+     * A constructor for the ITs with default parameters.
+     */
     public StreamableRDD(JavaRDD<T> rdd, OperationContext<?> context, UUID uuid, String clientHost, int clientPort, int batches, int batchSize) {
-        this(rdd, context, uuid, clientHost, clientPort, batches, batchSize, DEFAULT_PARALLEL_PARTITIONS);
+        this(rdd, context, uuid, clientHost, clientPort, batches, batchSize, DEFAULT_PARALLEL_PARTITIONS, OlapConfigurations.DEFAULT_SPARK_RESULT_STREAMING_THREADS);
     }
 
+
     public StreamableRDD(JavaRDD<T> rdd, OperationContext<?> context, UUID uuid, String clientHost, int clientPort,
-                         int batches, int batchSize, int parallelPartitions) {
+                         int batches, int batchSize, int parallelPartitions, int executorThreads) {
         this.rdd = rdd;
         this.context = context;
         this.uuid = uuid;
         this.host = clientHost;
         this.port = clientPort;
-        this.parallelPartitions = parallelPartitions % 2 == 0 ? parallelPartitions : parallelPartitions + 1;
-        this.executor = Executors.newFixedThreadPool(DEFAULT_PARALLEL_THREADS);
+        int defaultExecutorThreads = (int) max(min(ceil((float)parallelPartitions / 2), StreamableRDD.MAX_PARALLEL_THREADS),
+                OlapConfigurations.DEFAULT_SPARK_RESULT_STREAMING_THREADS);
+        this.executorThreads = max(defaultExecutorThreads, executorThreads);
+        this.parallelPartitions = this.executorThreads < parallelPartitions ? parallelPartitions : this.executorThreads;
+        this.executor = Executors.newFixedThreadPool(this.executorThreads);
```

Comment:
Could you provide a descriptive name of the threads in the thread pool to make it easy for debugging? something like `streamablerdd-streaming-thread-pool`

Reply:
@hatyo , I've added the name for pool and rename some variables to use 'stream' in names